### ISSUE TITLE
LibreELEC image: very similar to OpenELEC?

### ISSUE THREAD

**Issue by macmpi** [maxnet/berryboot#284]
Hi,

[LibreELEC](https://libreelec.tv/) is a recent fork of OpenELEC (Kodi), and I wanted to give it a shot under Berryboot.
It turns out the image is essentially made of a squashfs partition in the form of a file named SYSTEM within the base FAT partition. The secondary ext4 partition is almost empty.
Pretty much the same form than OpenELEC img actually.

Therefore I renamed that SYSTEM file into something like LibreELEC.img160, and installed it as-is on Berryboot: it seems to work great.
More testing would be needed, but if it is confirmed, it's a nice distribution that is very easy to use on BB!

Hope someone-else checks this out & confirm.
Thanks.

PS: as a good measure, it's better to unsquash it and re-squash it discarding /lib/modules folders as this is provided by Berryboot.
$ mkdir unsquashed_image_dir
$ cd unsquashed_image_dir
$ unsquashfs path_to/SYSTEM
$ sudo mksquashfs squashfs-root/ converted_image_for_berryboot.img -comp lzo -e lib/modules

**Comment by macmpi** [comment]
PS: as a good measure, it's better to unsquash it and re-squash it discarding /lib/modules folders as this is provided by Berryboot.
$ mkdir unsquashed_image_dir
$ cd unsquashed_image_dir
$ unsquashfs path_to/SYSTEM
$ sudo mksquashfs squashfs-root/ converted_image_for_berryboot.img -comp lzo -e lib/modules

**Comment by castaway1** [comment]
hello, can you upload your working LibreELEC image after unsquash please? im tinkering with converting LibreELEC 7.0 img for BB but cant get it. Thanks!

**Comment by agoldcheidt** [comment]
Yes, that method works, I got LibreELEC 7.0 image for berryboot tested and working. Download the image at: [http://berryboot.alexgoldcheidt.com/images/](http://berryboot.alexgoldcheidt.com/images/)

**Comment by muchachagrande** [comment]
Hi,
  I also found out how to update LibreELEC.
  Acording to Berryboot code, already working with OpenELEC, you have to copy SYSTEM file from inside the update .tar into /storage/.update.
  Then, at next reboot, Berryboot will find this image and replace its own with this new one before mounting it and booting.
  The result is a new version of LibreELEC with the aplied changes that you already made to the previous versions.
  This can be done because LibreELEC as well as it predecesor OpenELEC is distributed on a squashfs archive, the same that Berryboot relies on.
  The only flaw I found to this method is that you can't apply this kind of update to a restored backup of LibreELEC because when you do it, the main image becomes the restored backup and includes LibreELEC with all changed you made to your system. So, if after restoring a backup you update LibreELEC with this method you will be overwriting your customizations with a blank LibreELEC image and will lose your config.
  I hope you understand my poor english writing. Sorry for that.

Cheers

**Comment by macmpi** [comment]
I have just finished scripts to fully support LibreELEC in berryboot, including built-in software updates, resets, backup restore,...

From any Debian-based distribution (Raspbian under berryboot advised), run the following 1-line command: 
`curl -Ls --output installLibreELEC http://bit.do/InstLE; chmod u+x installLibreELEC; ./installLibreELEC`
and follow instructions to install LibreELEC image: it's that simple!

I wrote similar set of scripts for Volumio2: details [here](https://volumio.org/forum/multiboot-volumio2-with-kodi-under-berryboot-t6818.html#p33742).

Enjoy!

PS: code available [here](https://github.com/macmpi/berryboot-scripts)

**Comment by agoldcheidt** [comment]
That's amazing @macmpi I've add your scripts on my website as a useful resources.

**Comment by macmpi** [comment]
@agoldcheidt just to make sure there is no misunderstanding:
that 1-line command runs an installer which downloads a genuine LibreELEC image, patches-it, and installs it for use inside berryboot.

That patched image is then capable of supporting all LibreELEC features by itself, including auto-updates, resets, etc...
So that 1-line command is only needed (once) for initial image creation: such created image is then totally "autonomous", as would be a regular native LibreELEC install.

Full info [here](https://github.com/macmpi/berryboot-scripts).

PS: same concept applies for the similar Volumio2 command.
